### ISSUE TITLE
Improper Lighting and Reflections on iOS 

### ISSUE THREAD

**Issue by Raki** [google/filament#855]
I am trying to replicate "textured-object" android demo on iOS.

I compiled textured_pbr.mat to mobile platform as a header on mac.
I used the same image(png) assets from android.(baseColor,normalMap,aoRoughnessMetallic).
Generated environments on mac.

Using "hello-triangle" example as a reference I implemented the demo. But it seems something wrong with the lighting. I am getting output as below.

![img_0237](https://user-images.githubusercontent.com/824691/53148534-f7e59a00-35d0-11e9-8139-5a29ca3c730e.PNG)

Initially I suspected my tangent data. So  I tested this code(geometry generation part) on windows to validate the tangent data generation, there output is coming as expected with good light and reflections like below.

![testcase](https://user-images.githubusercontent.com/824691/53149039-8dcdf480-35d2-11e9-86d9-cb96d1d4ba42.png)

Please suggest me, other than the images and geometry is there anything I should check ? 

 - Device: iPhone 7
 - OS: 11.4.1

**Comment by romainguy** [comment]
Are you using the OpenGL backend on iOS?

**Comment by Raki** [comment]
@romainguy  Yes, I am using OpenGL backend

**Comment by bejado** [comment]
The hello-triangle demo has post processing turned off. Can you make sure you have it turned on?

```
filaView->setPostProcessingEnabled(true);
```

**Comment by romainguy** [comment]
Or just remove the line, post-process is on by default.

**Comment by Raki** [comment]
@bejado , @romainguy  I already tried that. I removed that line while setting up the filaView.
And tried with  `filaView->setPostProcessingEnabled(true);` too, still the same output

**Comment by bejado** [comment]
1. How are you importing the geometry on iOS?
2. Does the Windows demo have any lights added to the scene that aren't present in the iOS app?
3. Do you get the same results when switching out the model with another one? Specifically you could try models inside `assets/models`.

**Comment by Raki** [comment]
@bejado ,
1) I am generating the geometry at run time (procedurally).
2) Windows and iOS demo have same lights and environment.
3) So far requirement is rendering dynamic geometry with filament. I will test with static models from assets/models and update you on this.

Same geometry is working fine with Android and Windows filament demos.
Any chances of more demos for iOS in near future ?

**Comment by bejado** [comment]
Thanks- if the static models look the same across iOS / Windows, then the issue is probably with the tangent generation, though I don't know why this would be broken specifically on iOS. It'd be easier to help if you could host your project somewhere so I could download and try to reproduce the issue.

And yes, I have plans for more iOS demos. What types of feature demonstrations would you find helpful?

**Comment by Raki** [comment]
@bejado , I made changes to 'hello-pbr' iOS demo (with latest filament checkout) with my code.
Both static model and dynamic model are loading properly, so it seems there is no problem with tangent generation now.
Now I am getting proper output like the following.
![img_0240](https://user-images.githubusercontent.com/824691/53419747-675bef00-3a00-11e9-98f2-73f59360be1b.PNG)

Thank you so much for the new iOS demo and bug fixes. filament is awesome 👍  .

I am trying to capture framebuffer with transparent background, so I tried with `SwapChain::CONFIG_TRANSPARENT` , but it seems it was not implemented for iOS. I am getting black opaque background.

It would be more helpful if we have equal number of samples for both android and ios.
